transport-native-unix-common/src/main/java/io/netty/channel/unix/IovArray.java

```diff
@@ -52,23 +54,29 @@
      * The size of an {@code iovec} struct in bytes. This is calculated as we have 2 entries each of the size of the
      * address.
      */
-    private static final int IOV_SIZE = 2 * ADDRESS_SIZE;
+    public static final int IOV_SIZE = 2 * ADDRESS_SIZE;
 
     /**
      * The needed memory to hold up to {@code IOV_MAX} iov entries, where {@code IOV_MAX} signified
      * the maximum number of {@code iovec} structs that can be passed to {@code writev(...)}.
      */
-    private static final int CAPACITY = IOV_MAX * IOV_SIZE;
+    private static final int MAX_CAPACITY = IOV_MAX * IOV_SIZE;
 
-    private final ByteBuffer memory;
-    private final long memoryAddress;
+    private final ByteBuf memory;
     private int count;
     private long size;
     private long maxBytes = SSIZE_MAX;
 
     public IovArray() {
-        memory = Buffer.allocateDirectWithNativeOrder(CAPACITY);
-        memoryAddress = Buffer.memoryAddress(memory);
+        this(Unpooled.wrappedBuffer(Buffer.allocateDirectWithNativeOrder(MAX_CAPACITY)));
+    }
+
+    @SuppressWarnings("deprecation")
+    public IovArray(ByteBuf memory) {
+        assert memory.writerIndex() == 0;
+        assert memory.readerIndex() == 0;
+        this.memory = memory.order(
+                PlatformDependent.BIG_ENDIAN_NATIVE_ORDER ? ByteOrder.BIG_ENDIAN : ByteOrder.LITTLE_ENDIAN);
```

Comment:
only do this if `!PlatformDependent.hasUnsafe()`? otherwise we're unnecessarily allocating/wrapping on intel